OSPSuite.Presentation.Importer/Presenters/ImporterPresenter.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Assets;
+using OSPSuite.Core.Importer;
+using OSPSuite.Presentation.Core;
+using OSPSuite.Presentation.Importer.Core;
+using OSPSuite.Presentation.Importer.Services;
+using OSPSuite.Presentation.Importer.Views;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Core.Domain;
+using OSPSuite.Presentation.Importer.Core.DataFormat;
+
+namespace OSPSuite.Presentation.Importer.Presenters
+{ 
+   internal class ImporterPresenter : AbstractPresenter<IImporterView, IImporterPresenter>, IImporterPresenter
+   {
+      private readonly IDataViewingPresenter _dataViewingPresenter;
+      private readonly IColumnMappingPresenter _columnMappingPresenter;
+      private readonly ISourceFilePresenter _sourceFilePresenter;
+      private readonly IImporter _importer;
+      private readonly IApplicationController _applicationController;
+      private IDataSourceFile _dataSourceFile;
+      private IReadOnlyList<ColumnInfo> _columnInfos;
+      private DataImporterSettings _dataImporterSettings;
+
+      private IEnumerable<IDataFormat> _availableFormats;
+
+      public event FormatChangedHandler OnFormatChanged = delegate { };
+      public event ImportSingleSheetHandler OnImportSingleSheet;
+      public event ImportAllSheetsHandler OnImportAllSheets;
+      public event OnTriggerImportHandler OnTriggerImport = delegate { };
+
+      public ImporterPresenter
+      (
+         IImporterView view, 
+         IDataViewingPresenter dataViewingPresenter, 
+         IColumnMappingPresenter columnMappingPresenter, 
+         ISourceFilePresenter sourceFilePresenter,
+         IApplicationController applicationController,
+         IImporter importer
+      ) : base(view)
+      {
+         _importer = importer;
+         _applicationController = applicationController;
+         _view.AddDataViewingControl(dataViewingPresenter.View);
+         _view.AddColumnMappingControl(columnMappingPresenter.View);
+         _view.AddSourceFileControl(sourceFilePresenter.View);
+         _importer = importer;
+
+         _dataViewingPresenter = dataViewingPresenter;
+         _columnMappingPresenter = columnMappingPresenter;
+         _sourceFilePresenter = sourceFilePresenter;
+
+         _sourceFilePresenter.Title = Captions.Importer.PleaseSelectDataFile;
```

Comment:
why is this set from the outside?

Reply:
Because source file presenter is a reusable component for the selection of a file - can also be used in other context where the title should be different.